silq/tools/general_tools.py

```diff
@@ -735,3 +735,14 @@ def find_approximate_divisor(
             }
     else:
         return None
+        
+        
+def slice_length(s, length):
+    if isinstance(s, slice):
+        start = s.start if s.start is not None else 0
+        stop = s.stop if s.stop is not None else length
+        step = s.step if s.step is not None else 1
```

Comment:
At first I was confused why you seemingly had a variable back-up for length, I think a small docstring would be useful. I replaced `length` with `l` to make the docstring cleaner, but the code is a bit more terse.

```suggestion
def slice_length(s, l):
    """Returns the range of a slice `s` for an array of length `l`.
    """
    
    if isinstance(s, slice):
        start = s.start if s.start is not None else 0
        stop = s.stop if s.stop is not None else l
        step = s.step if s.step is not None else 1
```